### PR TITLE
prove EitherD/Either versions of init equivalent; init contract

### PR DESCRIPTION
**Pull request by haroldcarr**
prefix a LOT of stuff with `fake` to make it easier to identify and remove later

Impl/Handle
- prove EitherD/Either versions of initEMWithOutput are equivalent
- fake renames
- reordered stuff in file

Impl/Handle/InitProperties
- new file
- initial shot at contract

### CODE REVIEW

File: LibraBFT/Impl/Handle.agda [hunk 1]
```diff
@@ -123,10 +104,52 @@ runHandler st handler = ×-map₂ (outputsToActions {st}) (proj₂ (LBFT-run han
...
+initEMWithOutput≡ {x} iewo
+  with GenKeyFile.create 1 (0 ∷ 1 ∷ 2 ∷ 3 ∷ [])
+... | Right (nf , _ , vss , vv , pe , liws)
+  with Init.initialize 0 (nf , liws , vss , vv , pe) now ObmNeedFetch∙new pg
+... | Right y rewrite iewo = cong Right refl
+
+------------------------------------------------------------------------------
+-- TODO : ASK CHRIS : regarding EitherD-run
+
+zzz : EitherD ErrLog ℕ
```
Comment (mark-moir) [inline LibraBFT/Impl/Handle.agda]: Did you check in the right version of this?  This seems to be still in the state right after we figured out that we were confused over the difference between `initEMWithOutput ≡ Right x  and `EitherD-run initEMWithOutput ≡ Right x`.  We have one post-confusion case proved above, but not the 'Left` case, which should be equally trivial.  Also, I think we want (at least) the other direction...

Actually, I just pushed a commit to make clearer what I was about to say.  It shows that the proof that the two functions are equivalent is trivial, as expected/hoped.  All we have to do is the case analysis and rewrite/refl does the rest.

Comment (haroldcarr) [reply LibraBFT/Impl/Handle.agda]: Oh yeah - thanks for the update.

File: LibraBFT/Impl/Handle/InitProperties.agda [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    field
+       rmInv : Σ RoundManager
+                 λ rm → em ^∙ emProcessor ≡ just (RoundProcessorNormal rm)
+                      × Util.Invariants.RoundManagerInv rm
+
+  Contract : Either ErrLog (EpochManager × List Output) → Set
+  Contract (Left _)          = ⊤
+  Contract (Right (em , lo)) = ContractOk em lo
+
+--  open initEMWithOutputSpec
```
Comment (mark-moir) [inline LibraBFT/Impl/Handle/InitProperties.agda:30]: Might as well uncomment the postulated proof :smile:.

**Review by mark-moir — APPROVED** [review]
Sure, gahead :smile:.

File: LibraBFT/Impl/Handle/InitProperties.agda (hunk 2 above)
Comment (haroldcarr) [reply LibraBFT/Impl/Handle/InitProperties.agda:30]: It is not ready to be uncommented.

**Pull request merged**